### DISCUSSION TITLE
Can't connect to database using dbeaver

### DISCUSSION THREAD

**Discussion by Cristianooo** [planetscale/discussion#105] [Q&A] [answered]
I've been trying for a while to connect to my database through dbeaver. Can't seem to figure out the ssl connection. I'm on windows so I tried using [the curl bundle](https://curl.se/docs/caextract.html) as directed on your [docs](https://docs.planetscale.com/reference/secure-connections).

I always get the error _**Path does not chain with any of the trust anchors**_.
![error](https://user-images.githubusercontent.com/22159514/134468147-a89dd894-f9a9-41bc-850b-e4368e6e2098.JPG)

![cacert](https://user-images.githubusercontent.com/22159514/134468270-7c88aeb2-90cf-4536-9843-e324ca04608b.JPG)

**Comment by jgreet** [comment, marked as answer]
🤔 My understanding is that a recent version of the Java MySQL driver should fall back to using the system certificates by default. I don't entirely understand what's going on here, but if I set up the connection as I would expect to:

<img width="400" alt="CleanShot 2021-09-23 at 11 35 40@2x" src="https://user-images.githubusercontent.com/391331/134541037-3a5df68b-2e4c-40b4-a710-a545a3e0c340.png">
<img width="400" alt="CleanShot 2021-09-23 at 11 35 55@2x" src="https://user-images.githubusercontent.com/391331/134541046-57e76325-a912-429f-881a-cedd67571cbb.png">

And then set the `trustCertificateKeyStoreURL` to **any value and then set it back to empty**, the connection will succeed:

<img width="794" alt="CleanShot 2021-09-23 at 11 50 53@2x" src="https://user-images.githubusercontent.com/391331/134541055-eae3cc42-e6b4-4aa2-9fea-3b5344c74c0c.png">

This is Windows 10 and DBeaver 21.2.1.202109181446

**Reply by Cristianooo** [reply]
Thank you! that fixed it.

**Reply by iamhazel** [reply]
Thanks for the help!

**Reply by impe93** [reply]
Wow, thanks! Hacky solution but it worked for me lol (Mac user here)

**Reply by AnatoliiYarovyi** [reply]
Thank you so much!!! Helped me too (used Windows 10 and macOS 13.0.1)

**Reply by Jeypestana** [reply]
nice , solution efficent thabks

**Reply by adiguno** [reply]
Ty! works for macOS 13.0.1 as well

**Reply by jacobbroughton** [reply]
What in the world, such a strange solution but yeah this worked for me too. Thanks!

**Reply by Mestevam1976** [reply]
Thanks so much!!!Worked perfectly!

**Reply by kodlar** [reply]
I got ubuntu 22.0 and got the same error, thanks for weird solution

**Reply by luccasr73** [reply]
works on macOS ventura

**Reply by Lewiscowles1986** [reply]
So I had to brew install mysql-client, restart dbeaver, add a path to homebrew and then alter this setting, but then it worked 👍

**Reply by gnomeria** [reply]
It seems I need to do this all over again whenever I disconnect to the db. Anyway to fix it permanently?

**Reply by sss63232** [reply]
Thanks! Save my day🙏

**Reply by 0xPaulo** [reply]
I have no idea how you arrived at this solution, but it worked, thanks

**Reply by naishe** [reply]
Bizarre, but works! 🙏

**Reply by fwuffyboi** [reply]
looks like it works on all versions of software.
worked for me too on DBeaver 23.1.4, fedora 34.

**Reply by turtiesio** [reply]
You are AWESOME 👍👍👍👍

**Reply by paul-huszak** [reply]
This solves the `the trustAnchors parameter must be non-empty` problem too! 

Working on Mac M1 Pro running Sonoma 14.1.1. DBeaver version is 23.2.5.

Thanks a lot!

**Reply by zsiegel92** [reply]
Setting `trustCertificateKeyStoreURL` to any value and then back to empty did solve this for me.

Very weird! Thank you.

**Reply by audryhsu** [reply]
a true hero

**Comment by rijulsudhir** [comment]
On untick "Verify Server Certificate" it is working for me. I am sharing in case anyone else needs it.
![Screenshot_20221111_112831](https://user-images.githubusercontent.com/42400713/201274406-869c6d24-c9b4-41b1-8572-17737a5afab3.png)

**Reply by dbussink** [reply]
I'm going to hide this reply, since we do not want to recommend to anyone to run with certificate validation disabled, since it makes the connection not secure and vulnerable to man in the middle attacks.
